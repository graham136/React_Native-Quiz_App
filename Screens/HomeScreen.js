import * as React from 'react';
import {  StyleSheet, Text, View, Button } from 'react-native';
import { BackHandler } from 'react-native';

// Landing screen. Used to select the multiplication table
const HomeScreen = ({ navigation }) => {
    return (
        <View>
            <View style={styles.cardContainer}>
                <Text style={styles.headerText}>Multiplication</Text>
                <Text style={styles.headerText}>Tables</Text>
                <Text style={styles.questionText}>Please select a table?</Text>
                <View style={{ flexDirection: 'row' }}>
                    <View style={styles.button_1}>
                        <Button color='red' title='1' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 1}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='orange' title='2' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 2}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='yellow' title='3' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 3}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='green' title='4' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 4}) }} />
                    </View>
                </View>
                <View style={{ flexDirection: 'row' }}>
                    <View style={styles.button_1}>
                        <Button color='blue' title='5' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 5}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='indigo' title='6' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 6}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='violet' title='7' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 7}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='darkred' title='8' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 8}) }} />
                    </View>
                </View>
                <View style={{ flexDirection: 'row' }}>
                    <View style={styles.button_1}>
                        <Button color='darkorange' title='9' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 9}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='yellow' title='10' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 10}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='lightgreen' title='11' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 11}) }} />
                    </View>
                    <View style={styles.button_1}>
                        <Button color='purple' title='12' onPress={() => { navigation.navigate('Tables',{ multiplyTable: 12}) }} />
                    </View>
                </View>
                <View style={styles.button_2}>
                    <Button title='Exit' onPress={() => { BackHandler.exitApp(); }} />
                </View>
            </View>

        </View>
    );
}

export default HomeScreen;


// Styles
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        flexDirection: 'row',
    },
    cardContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        borderColor: 'grey',
        borderStyle: 'dashed',
        margin: 20,
        marginBottom: 20,
        marginTop: 20,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 5,
        },
        shadowOpacity: 0.36,
        shadowRadius: 6.68,
        elevation: 11,
    },
    contentContainer: {
        paddingTop: 30,
        marginVertical: 10,
    },
    headerText: {
        fontSize: 30,
        color: 'dodgerblue',
        textAlign: 'center',   
        padding: 2,     
    },
    
    questionText: {
        fontSize: 20,
        color: 'dodgerblue',
        textAlign: 'center',        
    },
    button_1: {
        width: '20%',
        height: 30,
        margin: 5,
        marginTop: 5,
        marginBottom: 25,
        alignSelf: 'center',
    },
    button_2: {
        width: '90%',
        height: 30,
        margin: 5,
        marginTop: 5,
        marginBottom: 25,
        alignSelf: 'center',
    },
});
