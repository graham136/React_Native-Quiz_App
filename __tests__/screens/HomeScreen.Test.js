import React from 'react';
import renderer from 'react-test-renderer';

import HomeScreen from '../../Screens/HomeScreen';

describe('<HomeScreen />', () => {
  
  // Snaphot test
  it('renders correctly', async () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});