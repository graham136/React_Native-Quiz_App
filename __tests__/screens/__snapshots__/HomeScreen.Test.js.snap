// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "grey",
        "borderStyle": "dashed",
        "elevation": 11,
        "justifyContent": "center",
        "margin": 20,
        "marginBottom": 20,
        "marginTop": 20,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.36,
        "shadowRadius": 6.68,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "dodgerblue",
          "fontSize": 30,
          "padding": 2,
          "textAlign": "center",
        }
      }
    >
      Multiplication
    </Text>
    <Text
      style={
        Object {
          "color": "dodgerblue",
          "fontSize": 30,
          "padding": 2,
          "textAlign": "center",
        }
      }
    >
      Tables
    </Text>
    <Text
      style={
        Object {
          "color": "dodgerblue",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Please select a table?
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "red",
                  },
                ]
              }
            >
              1
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "orange",
                  },
                ]
              }
            >
              2
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "yellow",
                  },
                ]
              }
            >
              3
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "green",
                  },
                ]
              }
            >
              4
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "blue",
                  },
                ]
              }
            >
              5
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "indigo",
                  },
                ]
              }
            >
              6
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "violet",
                  },
                ]
              }
            >
              7
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "darkred",
                  },
                ]
              }
            >
              8
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "darkorange",
                  },
                ]
              }
            >
              9
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "yellow",
                  },
                ]
              }
            >
              10
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "lightgreen",
                  },
                ]
              }
            >
              11
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "height": 30,
            "margin": 5,
            "marginBottom": 25,
            "marginTop": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "purple",
                  },
                ]
              }
            >
              12
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "height": 30,
          "margin": 5,
          "marginBottom": 25,
          "marginTop": 5,
          "width": "90%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Exit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
